### PR TITLE
Minor: Improve comments in row_hash.rs for skipping aggregation

### PR DESCRIPTION
**Pull request by alamb**
## Which issue does this PR close?

Follow on to https://github.com/apache/datafusion/pull/11627

## Rationale for this change

@2010YOUY01 had some good suggestions on https://github.com/apache/datafusion/pull/11627 such as https://github.com/apache/datafusion/pull/11627/files#r1695417159 which I wanted to capture as part of filing the follow up work

## What changes are included in this PR?
Update some comments

## Are there any user-facing changes?
Comment changes only, no functional changes

### CODE REVIEW

File: datafusion/physical-plan/src/aggregates/row_hash.rs [hunk 1]
```diff
@@ -100,22 +100,24 @@ struct SpillState {
...
     num_groups: usize,
 
-    /// Aggregation ratio check should be performed only when the
-    /// number of input rows exceeds this threshold
+    /// Aggregation ratio check performed when the number of input rows exceeds
+    /// this threshold (from `SessionConfig`)
     probe_rows_threshold: usize,
-    /// Maximum allowed value of `input_rows` / `num_groups` to
-    /// continue aggregation
+    /// Maximum ratio of `input_rows` / `num_groups` to continue aggregation
```
Comment (andygrove) [inline datafusion/physical-plan/src/aggregates/row_hash.rs:111]: The ratio seems to be stated inversely here?

```suggestion
    /// Maximum ratio of `num_groups` to `input_rows` for continuing aggregation
```
My understanding is based on the following code:

```rust
            self.should_skip = self.num_groups as f64 / self.input_rows as f64
                >= self.probe_ratio_threshold;
```

Comment (alamb) [reply datafusion/physical-plan/src/aggregates/row_hash.rs:111]: Yes, I agree -- thank you for the catch. @korowa is that your understanding too?

Comment (korowa) [reply datafusion/physical-plan/src/aggregates/row_hash.rs:111]: Yes, the numerator and denominator in this comment should be swapped (I've double checked `Config`, they are correct). Thank you!

File: datafusion/physical-plan/src/aggregates/row_hash.rs [hunk 2]
```diff
@@ -100,22 +100,24 @@ struct SpillState {
...
     probe_rows_threshold: usize,
-    /// Maximum allowed value of `input_rows` / `num_groups` to
-    /// continue aggregation
+    /// Maximum ratio of `num_groups` to `input_rows` for continuing aggregation
+    /// (from `SessionConfig`). If the ratio exceeds this value, aggregation
+    /// is skipped and input rows are directly converted to output
     probe_ratio_threshold: f64,
 
-    /// Flag indicating that further data aggregation mey be skipped
+    /// Flag indicating further data aggregation mey be skipped (decision made
```
Comment (korowa) [inline datafusion/physical-plan/src/aggregates/row_hash.rs:116]:
```suggestion
    /// Flag indicating further data aggregation may be skipped (decision made
```

File: datafusion/physical-plan/src/aggregates/row_hash.rs [hunk 3]
```diff
@@ -100,22 +100,24 @@ struct SpillState {
...
+    /// when probing complete)
     should_skip: bool,
-    /// Flag indicating that further updates of `SkipAggregationProbe`
-    /// state won't make any effect
+    /// Flag indicating further updates of `SkipAggregationProbe` state won't
+    /// make any effect (decision made when probing complete)
```
Comment (korowa) [inline datafusion/physical-plan/src/aggregates/row_hash.rs:120]:
```suggestion
    /// make any effect (set either while probing or on probing completion)
```

**Review by korowa — APPROVED** [review]
LGTM (just left a couple of suggestions).

Thank you @alamb @andygrove @2010YOUY01

**Review by jayzhan211 — APPROVED** [review]
👍

**Pull request merged**